### PR TITLE
improve: effects list sorting in material panel

### PR DESCRIPTION
**Pull request by doommm**
Re: #

### Changelog

* improve effects list sorting in material panel

image

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: editor/inspector/assets/material.js [hunk 1]
```diff
@@ -180,14 +180,18 @@ exports.methods = {
...
-            if (!effect.hideInEditor) {
-                arr.push(effect);
-            }
-            return arr;
-        }, []);
+        // see: https://developer.mozilla.org/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator
+        const collator = new Intl.Collator(undefined, {});
+
+        this.effects = Object.keys(effectMap)
+            .reduce((arr, uuid) => {
```
Comment (jareguo) [inline editor/inspector/assets/material.js]: 为什么不用 filter？好像更直观啊

File: editor/inspector/assets/material.js [hunk 2]
```diff
@@ -180,14 +180,18 @@ exports.methods = {
...
-
-        this.effects = Object.keys(effectMap).sort().reduce((arr, name) => {
-            const effect = effectMap[name];
-            if (!effect.hideInEditor) {
-                arr.push(effect);
-            }
-            return arr;
-        }, []);
+        // see: https://developer.mozilla.org/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator
+        const collator = new Intl.Collator(undefined, {});
```
Comment (jareguo) [inline editor/inspector/assets/material.js]: undefined 的话，应该传入 { numeric: true } 才对吧？

File: editor/inspector/assets/material.js (hunk 1 above)
Comment (doommm) [reply editor/inspector/assets/material.js]: 这里的 `reduce` 同时做了 `map`, `filter` 两件事，这个是单独的 `filter` 做不到的

File: editor/inspector/assets/material.js (hunk 2 above)
Comment (doommm) [reply editor/inspector/assets/material.js]: fixed

**Review by arsen2010 — APPROVED** [review]

File: editor/inspector/assets/material.js (hunk 1 above)
Comment (doommm) [reply editor/inspector/assets/material.js]: 改为使用 `Object.value(map).filter(predicate)`

**Review by arsen2010 — APPROVED** [review]

**Review by jareguo — APPROVED** [review]

**Pull request merged**